### PR TITLE
Return exception code as Android does to be consistent. 

### PR DESCRIPTION
**Pull request by mehmetf**
Do not return error if user cancels signin on iOS.

### CODE REVIEW

File: packages/google_sign_in/ios/Classes/GoogleSignInPlugin.m [hunk 1]
```diff
@@ -120,8 +120,9 @@ - (void)signIn:(GIDSignIn *)signIn
     didSignInForUser:(GIDGoogleUser *)user
            withError:(NSError *)error {
   if (error != nil) {
-    if (error.code == -4) {
-      // Occurs when silent sign-in is not possible, return an empty user in this case
+    if (error.code == -4 || error.code == -5) {
```
Comment (goderbauer) [inline packages/google_sign_in/ios/Classes/GoogleSignInPlugin.m]: Are there no constants defined for these error codes? If not, can we define some to make this more readable?

File: packages/google_sign_in/android/src/main/java/io/flutter/plugins/googlesignin/GoogleSignInPlugin.java [hunk 2]
```diff
@@ -398,7 +397,8 @@ public boolean onActivityResult(int requestCode, int resultCode, Intent data) {
     }
 
     if (resultCode != Activity.RESULT_OK) {
-      finishWithError(ERROR_REASON_CANCELED, String.valueOf(resultCode));
+      // User cancelled so we return a null auth object.
```
Comment (goderbauer) [inline packages/google_sign_in/android/src/main/java/io/flutter/plugins/googlesignin/GoogleSignInPlugin.java]: Should this only happen if `resultCode == Activity.RESULT_CANCELED`? (Not sure what other failure reasons could be...)

**Comment by goderbauer** [comment]
The user can also cancel out of some steps during the `init` step (e.g. if they are asked to update Google Play Services, they can decline that). These currently result in a `ERROR_REASON_CONNECTION_FAILED` exception. Should those - for consistency (the user of this plugin doesn't really now that a sign-in call results in two steps: `init` and the actual `signIn`) - also succeed with `null`?

**Comment by mehmetf** [comment]
This is submitted by @goderbauer

**Pull request closed without merging**